Mage.Sets/src/mage/cards/t/ThunderbladeCharge.java

```diff
@@ -34,10 +36,12 @@ public ThunderbladeCharge(UUID ownerId, CardSetInfo setInfo) {
         // Whenever one or more creatures you control deal combat damage to a player, 
         // if Thunderblade Charge is in your graveyard, you may pay {2}{R}{R}{R}. 
         // If you do, you may cast it without paying its mana cost.
-        this.addAbility(new DealCombatDamageControlledTriggeredAbility(Zone.GRAVEYARD,
-                new DoIfCostPaid(new ThunderbladeChargeCastEffect(), new ManaCostsImpl<>("{2}{R}{R}{R}"))
-                        .setText("if {this} is in your graveyard, you may pay {2}{R}{R}{R}. "
-                                + "If you do, you may cast it without paying its mana cost")));
+        this.addAbility(new DealCombatDamageControlledTriggeredAbility(
+            Zone.GRAVEYARD,
+            new DoIfCostPaid(new ThunderbladeChargeCastEffect(), new ManaCostsImpl<>("{2}{R}{R}{R}"))
+                    .setText("if {this} is in your graveyard, you may pay {2}{R}{R}{R}. If you do, you may cast it without paying its mana cost"),
+            StaticFilters.FILTER_PERMANENT_CREATURES, SetTargetPointer.NONE, true)
```

Comment:
DoIfCostPaid already contains an option, so you don't want triggers using it to be optional